docs/source/contributing.md

```diff
@@ -0,0 +1,221 @@
+echopype makes extensive use of GitHub Actions for continuous integration (CI)
+of unit tests and other code quality controls. Every pull request (PR) triggers the CI.
+See `echopype/.github/workflows <https://github.com/OSOceanAcoustics/echopype/tree/main/.github/workflows>`_,
+especially `pr.yaml <https://github.com/OSOceanAcoustics/echopype/blob/main/.github/workflows/pr.yaml>`_.
+
+The entire test suite can be a bit slow, taking up to 40 minutes or more.
+To mitigate this, the CI default is to run tests only for subpackages that
+were modified in the PR; this is done via ``.ci_helpers/run-test.py``
+(see the `Running the tests`_ section). To have the CI execute the
+entire test suite, add the string "[all tests ci]" to the PR title.
+Under special circumstances, when the submitted changes have a
+very limited scope (such as contributions to the documentation)
+or you know exactly what you're doing
+(you're a seasoned echopype contributor), the CI can be skipped.
+This is done by adding the string "[skip ci]" to the PR title. -->
+
+
+
+## Documentation
+
+### Function and object docstrings
+
+For inline strings documenting functions and objects ("docstrings"),
+we use the [numpydoc style](https://numpydoc.readthedocs.io/en/latest/format.html) (Numpy docstring format).
+
+
+### General setup
+
+Echopype documentation (https://echopype.readthedocs.io) is based on [Jupyter Book](https://jupyterbook.org/en/stable/intro.html),
+which are rendered under the hood with [Sphinx](https://www.sphinx-doc.org).
+The documentation is hosted on [Read The Docs](https://readthedocs.org).
+
+To build the documentation locally, run:
+```shell
+jupyter-book build docs/source --path-output docs
+```
+
+To view the HTML files generated by Jupyter Book, open `docs/_build/html/index.html` in your browser.
+
+For some quick orientation of where things are:
+- The documentation package dependencies are in `docs/requirements.txt`
+- The documentation source files are in the `docs/source` directory
+- The Jupyter Book [configurations](https://jupyterbook.org/en/stable/customize/config.html)
+  is in `docs/source/_config.yml`
+- The [table of contents](https://jupyterbook.org/en/stable/structure/toc.html) for the sidebar
+  is in `docs/source/_toc.yml`
+
+### Versions
+
+ReadTheDocs defaults to having the `stable` version tracking the most recent release and the `main` version tracking the latest changes in the `main` branch of the repository. We follow this pattern for our documentation. See [RTD Versions](https://docs.readthedocs.io/en/stable/versions.html) for more information.
+
+
+
+## GitHub Actions for continuous integration (CI)
+When a PR is created, the CI will run through all tests, basic spelling and formatting checks
+(via pre-commit), and build the documentation.
+You can check the test results in a section at the bottom of the PR like below:
+![ci_runs](https://user-images.githubusercontent.com/15334215/165646718-ebfd4041-b110-4b54-a5b9-54a7a08bc982.png)
```

Comment:
This might just be my slow wifi, but this image loaded pretty slowly. Is this a large png? Also, just curious, do these 'githubusercontent' png links last forever?

Reply:
it seems to last forever. Maybe I just save a smaller version in the repo.